beaker/session.py

```diff
@@ -366,14 +458,106 @@ def invalidate(self):
 
     def load(self):
         "Loads the data from this session from persistent storage"
-        self.namespace = self.namespace_class(self.id,
-            data_dir=self.data_dir,
-            digest_filenames=False,
-            **self.namespace_args)
+        if self.old_reads():
+            self.namespace = self.namespace_class(self.id,
+                data_dir=self.data_dir,
+                digest_filenames=False,
+                **self.namespace_args)
+
+        # REMOVE AFTER MIGRATION
+        if self.new_reads():
+            log.info("Creating namespace2 for reads")
+            self.namespace2 = self.namespace_class(self.id,
+                data_dir=self.data_dir,
+                digest_filenames=False,
+                column_family=self.migration_provider().column_family,
+                **self.namespace_args)
+        # END REMOVE AFTER MIGRATION
+
         now = time.time()
         if self.use_cookies:
             self.request['set_cookie'] = True
 
+        timed_out = False
+        read_value = False
+        session_data = None
+        # REMOVE AFTER MIGRATION - Attempt to read from the new backend
+        if self.new_reads():
+            self.namespace2.acquire_read_lock()
+            try:
+                self.clear()
+                try:
+                    session_data = self.namespace2['session']
+
+                    if (session_data is not None and self.encrypt_key):
+                        session_data = self._decrypt_data(session_data, migration=True)
+                    elif not self.old_reads():
+                        # Memcached always returns a key, its None when its not
+                        # present
+                        session_data = {
+                            '_creation_time': now,
+                            '_accessed_time': now
+                        }
+                        self.is_new = True
+                        read_value = True
+
+                except (KeyError, TypeError):
+                    session_data = None
+                    if self.old_reads():
+                        # We still have another backend we could be reading from, so don't create new sessions here
+                        pass
+                    else:
+                        # Post migration we should 
+                        session_data = {
+                            '_creation_time': now,
+                            '_accessed_time': now
+                        }
+                        self.is_new = True
+                        read_value = True
+
+                if not self.old_reads() and (session_data is None or len(session_data) == 0):
+                    session_data = {
```

Comment:
you're right btw, this construct appearing 3 times in the code is really weird. it's hard to follow what is happening, however, I see this is all a copy paste from the original beaker code, you're simply not doing it when old reads are enabled which is :+1: